File: MH/DAY6/10828.js
const path = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
const input = require("fs").readFileSync(path).toString().split("\n");
const T = input.shift();
const st = [];

function stack(input) {
  if (input.includes("push")) {
    st.push(+input.slice(5));
  }
  if (input.includes("pop")) {
    if (st.length === 0) return -1;
    return st.pop();
  }
  if (input.includes("size")) {
    return st.length;
  }
  if (input.includes("empty")) {
    return st.length === 0 ? 1 : 0;
  }
  if (input.includes("top")) {
    if (st.length === 0) return -1;
    return st[st.length - 1];
  }
}

let answer = [];
for (let i = 0; i < T; i++) {
  const result = stack(input[i]);
  if (result !== undefined) {
    answer.push(result);
  }
}

console.log(answer.join('\n'));

File: SY/stack/10828.py
n=int(input())
li=[input() for _ in range(n)]
stack=[]

for i in li:
	if 'push' in i:
		stack.append(int(i[5:]))
	elif i=='pop':
		if not stack:
			print(-1)
		else:
			print(stack.pop())
	elif i=='size':
		print(len(stack))
	elif i=='empty':
		if not stack:
			print(1)
		else:
			print(0)
	else:
		if not stack:
			print(-1)
		else:
			print(stack[-1])
